### ISSUE TITLE
Build failure with ECL 21.2.1

### ISSUE THREAD

**Issue by sternenseemann** [pcostanza/closer-mop#20]
```

;;; Loading "/nix/store/y3fmiqvsmpd2zqnznsgkm1yafak94jvh-disable-debugger.lisp"

;;; Loading #P"/nix/store/n0cyh2wvsjwd012s68dv0w9j14h1bd8m-ecl-21.2.1/lib/ecl-21.2.1/cmp.fasc"

;;; Compiling /nix/store/ingdl8krq7p3mdcvzjf4n07kfw39bf35-lisp-closer-mop-20211209-git/closer-mop-packages.lisp.

;;; Compiling #<input stream /nix/store/ingdl8krq7p3mdcvzjf4n07kfw39bf35-lisp-closer-mop-20211209-git/closer-mop-packages.lisp 0x7ffff783ce60>.

;;; Loading "//build/_nix_store_ingdl8krq7p3mdcvzjf4n07kfw39bf35-lisp-closer-mop-20211209-git_closer-mop-packages.lisp.fasc"

;;; Compiling /nix/store/ingdl8krq7p3mdcvzjf4n07kfw39bf35-lisp-closer-mop-20211209-git/closer-mop-shared.lisp.

;;; Compiling #<input stream /nix/store/ingdl8krq7p3mdcvzjf4n07kfw39bf35-lisp-closer-mop-20211209-git/closer-mop-shared.lisp 0x7ffff6fd0d20>.

;;; Loading "//build/_nix_store_ingdl8krq7p3mdcvzjf4n07kfw39bf35-lisp-closer-mop-20211209-git_closer-mop-shared.lisp.fasc"

;;; Compiling /nix/store/ingdl8krq7p3mdcvzjf4n07kfw39bf35-lisp-closer-mop-20211209-git/closer-ecl.lisp.

;;; Compiling #<input stream /nix/store/ingdl8krq7p3mdcvzjf4n07kfw39bf35-lisp-closer-mop-20211209-git/closer-ecl.lisp 0x7ffff6f0a8c0>.

;;; Loading "//build/_nix_store_ingdl8krq7p3mdcvzjf4n07kfw39bf35-lisp-closer-mop-20211209-git_closer-ecl.lisp.fasc"

;;;

;;; Compiling /nix/store/ingdl8krq7p3mdcvzjf4n07kfw39bf35-lisp-closer-mop-20211209-git/closer-mop-packages.lisp.

;;; OPTIMIZE levels: Safety=2, Space=0, Speed=3, Debug=0

;;;

;;; End of Pass 1.

;;; Finished compiling /nix/store/ingdl8krq7p3mdcvzjf4n07kfw39bf35-lisp-closer-mop-20211209-git/closer-mop-packages.lisp.

;;;

;;;

;;; Compiling /nix/store/ingdl8krq7p3mdcvzjf4n07kfw39bf35-lisp-closer-mop-20211209-git/closer-mop-shared.lisp.

;;; OPTIMIZE levels: Safety=2, Space=0, Speed=3, Debug=0

;;;

;;; Compiling (DEFUN REQUIRED-ARGS ...).

;;; Compiling (DEFUN ENSURE-FINALIZED ...).

;;; Compiling (DEFUN SUBCLASSP ...).

;;; Compiling (DEFCLASS STANDARD-GENERIC-FUNCTION ...).

;;; Compiling (DEFUN COMPUTE-ARGUMENT-ORDER ...).

;;; Compiling (DEFUN PARSE-METHOD-BODY ...).

;;; Compiling (DEFUN BLOCK-NAME ...).

;;; Internal error:

;;;   in file closer-mop-shared.lisp, position 2299

;;;   at (DEFUN BLOCK-NAME ...)

;;;   ** In function CAR, the value of the only argument is

;;;   0

;;; which is not of the expected type CONS
```

**Comment by pcostanza** [comment]
I cannot reproduce this. When I compile closer-mop on my Mac with ECL 21.2.1, I don't see any problems.

What is nix/store? What are the disable-debugger.lisp and cmp.fasc files that are loaded in the beginning? What platform are you doing this on? What commands do I need to enter to get the same result?

Thanks,
Pascal

**Comment by sternenseemann** [comment]
Seems like this was a bug in ECL's bytecode compiler which has been fixed at some point between 21.2.1 and current HEAD (`1c989247c1b0bf1d38a76aec30b9ca5e41afe1e3`). For the record the setup I was using was roughly:

* ECL 21.2.1 + patches from master allowing bytecode `fasc` files that had been concatenated from multiple ones to be loaded again
* Creating a static library while relying on the bytecode to be able to load it (since ECL can't load the static libs it produces)

If anyone runs into something similar, I'd recommend to use ECL from git, I ran into the same compilation failure with recent ironclad as well.
